server/src/main/java/com/google/fhir/gateway/FhirProxyServer.java

```diff
@@ -95,35 +65,20 @@ protected void initialize() throws ServletException {
     try {
       logger.info("Adding BearerAuthorizationInterceptor ");
       AccessCheckerFactory checkerFactory = chooseAccessCheckerFactory();
-      HttpFhirClient httpFhirClient = chooseHttpFhirClient(backendType, fhirStore);
+      HttpFhirClient httpFhirClient = FhirClientFactory.createFhirClientFromEnvVars();
```

Comment:
You are right but there are two points to consider:
- `HttpFhirClient` and `TokenVerifier` are not necessarily singletons; I mean for example if for whatever reason the custom endpoint need to talk to a different IDP or FHIR-server, it should be able to reuse these classes (although that may need some code tweaks).
- The best way of doing this is probably through dependency injection but I did not want to bring a DI framework (e.g., Spring) into the examples. I have even tried to avoid it to a good extent in the `server` code. I think if people want to enforce the singleton pattern in their apps, then can do that with DI.